File: backend/src/models/user.model.ts
import type { Request } from "express";

export interface User {
    id: string;
    firstName: string;
    lastName: string;
    email: string;
    passwordHash: string;
    phone: string;
}
export interface RegisterUser {
    firstName: string;
    lastName: string;
    email: string;
    password: string;
    phone: string;
}
export interface RequestUser extends Request {
    user?: {
        id: string,
    }
}

File: yonetim_paneli/lib/core/models/user.model.dart
class User {
  final String firstName;
  final String lastName;
  final String email;
  final String phone;
  User({
    required this.firstName,
    required this.lastName,
    required this.email,
    required this.phone,
  });
  factory User.fromJson(Map<String, dynamic> json) {
    return User(
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'] ?? '',
      email: json['email'] ?? '',
      phone: json['phone'] ?? '',
    );
  }
}
